pkg/cmd/image/save.go

```diff
@@ -0,0 +1,90 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package image
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/images/archive"
+	"github.com/containerd/nerdctl/pkg/api/types"
+	"github.com/containerd/nerdctl/pkg/idutil/imagewalker"
+	"github.com/containerd/nerdctl/pkg/platformutil"
+	"github.com/containerd/nerdctl/pkg/strutil"
+	"github.com/mattn/go-isatty"
+)
+
+func Save(ctx context.Context, client *containerd.Client, images []string, options types.ImageSaveOptions) error {
+	writer := options.Stdout
+	if options.Output != "" {
+		f, err := os.OpenFile(options.Output, os.O_CREATE|os.O_WRONLY, 0644)
+		if err != nil {
+			return err
+		}
+		defer f.Close()
+		writer = f
+	} else {
+		if isatty.IsTerminal(os.Stdout.Fd()) {
+			return fmt.Errorf("cowardly refusing to save to a terminal. Use the -o flag or redirect")
+		}
+	}
+
+	return saveImages(ctx, client, images, writer, options)
+}
+
+func saveImages(ctx context.Context, client *containerd.Client, images []string, writer io.Writer, opt types.ImageSaveOptions, exportOpts ...archive.ExportOpt) error {
```

Comment:
Can we also expose this func? It makes sense to me that caller has own `io.writer` and call this func directly (i.e., they don't need us create file writer for them, as in `Save`)

Reply:
Makes sense. Currently, there is only one reference to it, so I kept it not exposed.